cli/command/trust/signer_add.go

```diff
@@ -0,0 +1,141 @@
+package trust
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/image"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/cli/opts"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	tufutils "github.com/docker/notary/tuf/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type signerAddOptions struct {
+	keys   opts.ListOpts
+	signer string
+	images []string
+}
+
+func newSignerAddCommand(dockerCli command.Cli) *cobra.Command {
+	var options signerAddOptions
+	cmd := &cobra.Command{
+		Use:   "add OPTIONS NAME IMAGE [IMAGE...] ",
+		Short: "Add a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return addSigner(dockerCli, options)
+		},
+	}
+	flags := cmd.Flags()
+	options.keys = opts.NewListOpts(nil)
+	flags.Var(&options.keys, "key", "Path to the signer's public key file(s)")
+	return cmd
+}
+
+var validSignerName = regexp.MustCompile(`^[a-z0-9][a-z0-9\_\-]*$`).MatchString
+
+func addSigner(cli command.Cli, options signerAddOptions) error {
+	signerName := options.signer
+	if !validSignerName(signerName) {
+		return fmt.Errorf("signer name \"%s\" must start with lowercase alphanumeric characters and can include \"-\" or \"_\" after the first character", signerName)
+	}
+	if signerName == "releases" {
+		return fmt.Errorf("releases is a reserved keyword, please use a different signer name")
+	}
+
+	if options.keys.Len() == 0 {
+		return fmt.Errorf("path to a public key must be provided using the `--key` flag")
+	}
+	signerPubKeys, err := ingestPublicKeys(options.keys.GetAll())
+	if err != nil {
+		return err
+	}
+	var errImages []string
+	for _, imageName := range options.images {
+		fmt.Fprintf(cli.Out(), "Adding signer \"%s\" to %s...\n", signerName, imageName)
+		if err := addSignerToImage(cli, signerName, imageName, signerPubKeys); err != nil {
+			fmt.Fprintln(cli.Err(), err.Error()+"\n")
```

Comment:
`Fprintln()` already adds a newline, but there's also a `\n` at the end of the message. Are you sure we want an extra blank line between these?

Reply:
I added an extra blank line to separate the signer output if there are multiple ex:
```
$ docker trust signer add alice example/trust-demo example/trust-demo2 --key alice.crt
Adding signer "alice" to example/trust-demo...
Enter passphrase for repository key with ID 95b9e55: 
Successfully added signer: alice to example/trust-demo

Adding signer "alice" to example/trust-demo2...
Enter passphrase for repository key with ID ece554f: 
Successfully added signer: alice to example/trust-demo2
```

Let me know if you'd prefer for the blank line to be removed, but I think it helps improve readability

Reply:
Ok